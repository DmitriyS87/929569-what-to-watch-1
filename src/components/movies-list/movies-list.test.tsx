import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MoviesList from './movies-list';
import TEST_MOCKS from '../../mocks/test-mocks.js';

it(`MoviesList correctly renders after relunch`, () => {
  const mockHandle = jest.fn();
  const tree = renderer
    .create(<MoviesList
      movies={TEST_MOCKS.films}
      setActive={mockHandle}
      activeGenre={TEST_MOCKS.activeGenre}
      activeItem={null}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
