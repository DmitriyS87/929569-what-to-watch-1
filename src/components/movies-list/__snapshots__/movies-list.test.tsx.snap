// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList correctly renders after relunch 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Aviator"
        height="175"
        src="img/aviator.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/:0"
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Pulp Fiction"
        height="175"
        src="img/pulp-fiction.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/:1"
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Aviator CLONE"
        height="175"
        src="img/aviator.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/:2"
      >
        Aviator CLONE
      </a>
    </h3>
  </article>
</div>
`;
